pymatgen/io/openmm/utils.py

```diff
@@ -360,111 +393,125 @@ def parameterize_system(
-        # TODO: figure out how to add partial charges
-        small_ffs = [
-            "smirnoff99Frosst-1.0.2",
-            "smirnoff99Frosst-1.0.0",
-            "smirnoff99Frosst-1.1.0",
-            "smirnoff99Frosst-1.0.4",
-            "smirnoff99Frosst-1.0.8",
-            "smirnoff99Frosst-1.0.6",
-            "smirnoff99Frosst-1.0.3",
-            "smirnoff99Frosst-1.0.1",
-            "smirnoff99Frosst-1.0.5",
-            "smirnoff99Frosst-1.0.9",
-            "smirnoff99Frosst-1.0.7",
-            "openff-1.0.1",
-            "openff-1.1.1",
-            "openff-1.0.0-RC1",
-            "openff-1.2.0",
-            "openff-1.3.0",
-            "openff-2.0.0-rc.2",
-            "openff-2.0.0",
-            "openff-1.1.0",
-            "openff-1.0.0",
-            "openff-1.0.0-RC2",
-            "openff-1.3.1",
-            "openff-1.2.1",
-            "openff-1.3.1-alpha.1",
-            "openff-2.0.0-rc.1",
-            "gaff-1.4",
-            "gaff-1.8",
-            "gaff-1.81",
-            "gaff-2.1",
-            "gaff-2.11",
-        ]
         small_molecules = {}
         large_or_water = {}
+        # iterate through each smile, if no forcefielded provided use Sage
         # iterate through each molecule and forcefield input as list
-        for smile, ff_name in force_field.items():
-            openff_mol = openff.toolkit.topology.Molecule.from_smiles(smile)
-            # Assign mols and forcefield as small molecule vs AMBER or
-            # CHARMM
-            if ff_name.lower() in small_ffs:
-                small_molecules[openff_mol] = ff_name.lower()
+        # Add charges to the molecule if provided
+        charged_mols = assign_charges_to_mols(
+            smile_strings=smile_strings,
+            partial_charges=partial_charges,
+            partial_charge_method=partial_charge_method,
+            partial_charge_scaling=partial_charge_scaling,
+        )
+        for smile, charged_mol in zip(smile_strings, charged_mols):
+            if smile in force_field.keys():
+                ff_name = force_field[smile]
+                if ff_name.lower() in all_small_ffs or ff_name.lower() in basic_small_ffs:
+                    small_molecules[charged_mol] = ff_name
+                else:
+                    large_or_water[charged_mol] = ff_name
             else:
-                large_or_water[openff_mol] = ff_name.lower()
-        forcefield_omm = omm_ForceField()
+                small_molecules[charged_mol] = "sage"
+
+        # Determine category of forcefield for deciding which water model
+        large_ff_category = None
+        for ff in large_or_water.values():
+            temp_ff_string = None
+            if "amber" in ff.lower():
+                temp_ff_string = "amber"
+            elif "charmm" in ff.lower():
+                temp_ff_string = "charmm"
+            elif "amoeba" in ff.lower():
+                temp_ff_string = "amoeba"
+            else:
+                continue
```

Comment:
This is to see what forcefield to use in general in order to assign the proper water model. For example, if the user inputs "tip3p," we need to know if the rest of the simulation uses "amber" or "charmm", because they each have their own tip3p: "amber14/tip3p.xml" vs "charmm36/water.xml".
The `continue` is meant to skip assigning a `large_ff_category` if the forcefield specified is a water model. It could be simpler to keep two separate lists, but I was trying to stay consistent with the OpenMM dichotomy

Reply:
Also updated it to not use `continue` anyway